### PR TITLE
refactor(GuildChannel)!: default setParent's lockPermissions to false

### PR DESCRIPTION
**Pull request by ImRodry**
**Please describe the changes this PR makes and why it should be merged:**

This PR changes the default for lockPermissions on the GuildChannel#setParent method to false and also fixes the documentation. The reasoning for this change is that, when using this method, users are most likely only expecting to change the parent of the channel, as happens with all the other setX methods on the library as a whole. Giving them the permission to lockPermissions is nice but doing this silently might result in unintentional changes (has happened to me countless times) and I believe it's better to let the user explicitly add this in, but I'm open to feedback if you guys disagree

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

**Comment by Idris1401** [comment]
When we create a channel on a category Discord is automatically syncing the permissions or when we move a channel to a category Discord puts more emphasis on the sync button so I would say that we should keep it to `true`.

**Comment by ImRodry** [comment]
I know that, but my argument is simply that these set methods are supposed to change only what is in their name. Also even though the sync permissions button is given more emphasis on the client, it still required explicit action from the user, which is what I’m trying to do here. The goal here is to mainly prevent accidental changes for users who are unaware of this default

**Review by vladfrangu — APPROVED** [review]

**Review by Jiralite — APPROVED** [review]

File: packages/discord.js/src/structures/GuildChannel.js [hunk 1]
```diff
@@ -327,11 +327,16 @@ class GuildChannel extends BaseChannel {
...
+   *   .then(channel => console.log(`New parent of ${channel.name}: ${channel.parent.name}`))
+   *   .catch(console.error);
+   * @example
+   * // Move a channel and sync its permissions with the parent
+   * message.channel.setParent('355908108431917066', { lockPermissions: true })
+   *   .then(channel => console.log(`Moved ${message.channel.name} to ${channel.parent.name}`))
    *   .catch(console.error);
    */
-  setParent(channel, { lockPermissions = true, reason } = {}) {
+  setParent(channel, { lockPermissions = false, reason } = {}) {
```
Comment (almeidx) [inline packages/discord.js/src/structures/GuildChannel.js:339]: What's Discord's default behaviour? Do we need this default value at all?

Comment (ImRodry) [reply packages/discord.js/src/structures/GuildChannel.js:339]: The default API behavior is what this PR is changing. Setting lockPermissions to true changes the permissionOverwrites as calculated by djs, setting it to false only changes the parent and nothing else. On the client, it usually prompts the user whether they want to sync or not.
In summary: false is the default behavior

**Review by kyranet — APPROVED** [review]

**Pull request merged**